redis-lib/src/main/java/com/streamsets/pipeline/stage/origin/redis/RedisBatchReaderSource.java

```
@@ -0,0 +1,141 @@
+
+            redisClient.scriptFlush();
+            redisClient.disconnect();
+            redisClient.close();
+        }
+
+        // If issues is not empty, the UI will inform the user of each configuration issue in the list.
+        return issues;
+    }
+
+    @Override
+    public void destroy() {
+        redisClient.scriptFlush();
+
+        super.destroy();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String produce(String lastSourceOffset, int maxBatchSize, BatchMaker batchMaker) throws StageException {
+        createRedisClient();
+
+        // Offsets can vary depending on the data source. Here we use an integer as an example only.
+        long nextSourceOffset = 0;
+        if (lastSourceOffset != null) {
+            nextSourceOffset = Long.parseLong(lastSourceOffset);
+        }
+
+        ArrayList<String> keys = new ArrayList<>();
+        keys.add(queueName);
+        ArrayList<String> args = new ArrayList<>();
+        args.add(Integer.toString(maxBatchSize));
+        HashSet<String> rawResult = new HashSet<>();
+        try {
+            rawResult = new HashSet<>((List<String>) redisClient.evalsha(shaKey, keys, args));
+        } catch (JedisDataException e) {
+            if (e.getMessage().startsWith("NOSCRIPT")) {
+                // script unregistered by 3rd party
+                LOG.error(e.getMessage(), e);
+                throw e;
+            }
+            LOG.error(e.getMessage(), e);
+        }
+
+        for (String message : rawResult) {
+            List<Record> records = processRedisMessage("id::" + nextSourceOffset, message);
+            for (Record record : records) {
+                batchMaker.addRecord(record);
+            }
+            ++nextSourceOffset;
+        }
+
+        return lastSourceOffset;
+    }
+
+    private void createRedisClient() {
+        List<ConfigIssue> issues = new ArrayList<>();
+        if (createRedisClient(issues)) {
```

Shouldn't we report an error or throw an exception if the createRedisClient will return false here? It doesn't make sense to continue processing the messages, right?